sklearn/semi_supervised/_label_propagation.py

```diff
@@ -326,6 +326,33 @@ def fit(self, X, y):
         self.transduction_ = transduction.ravel()
         return self
 
+    def fit_predict(self, X, y):
+        """Fit semi-supervised label propagation to X and perform inductive inference.
+
+        The input samples (labeled and unlabeled) are provided by matrix X,
+        and target labels are provided by matrix y. We conventionally apply the
+        label -1 to unlabeled samples in matrix y in a semi-supervised
+        classification.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data, where `n_samples` is the number of samples
+            and `n_features` is the number of features.
+
+        y : array-like of shape (n_samples,)
+            Target class values with unlabeled points marked as -1.
+            All unlabeled samples will be transductively assigned labels
+            internally.
+
+        Returns
+        -------
+        y : ndarray of shape (n_samples,)
+            Predictions for input data.
+        """
+        self.fit(X, y)
+        return self.transduction_
```

Comment:
`transduction_` is already documented in `LabelPropagation` as:

```
    transduction_ : ndarray of shape (n_samples)
        Label assigned to each item via the transduction.
```

We can improve that description if you think it's not clear, and we can also add `attributes` to `BaseLabelPropagation` for clarity maybe. I think I might prefer that to implementing `fit_predict`.

Reply:
The reason I thought this should exist is that similar methods exist for other sklearn classes, e.g. [`KMeans.fit_predict`](https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/cluster/_kmeans.py#L1007) does basically the same thing.

Reply:
Yes, but I'm not sure if that's a good thing.

cc @scikit-learn/core-devs

Reply:
It definitely feels (is) a bit redundant, but I think the benefit of a more "consistent" API is worth it. It helps newcomers infer usage across estimators. Overall I'm in favor.